modules/unsupported/csv/src/main/java/org/geotools/data/csv/parse/CSVLatLonStrategy.java

```diff
@@ -65,26 +76,93 @@ protected SimpleFeatureType buildFeatureType() {
             Class<?> type = typesFromData.get(col);
             if (isLatitude(col)) {
                 latSpelling = col;
-                if (CSVStrategySupport.isNumeric(type)) {
+                if (CSVStrategy.isNumeric(type)) {
                     validLat = true;
                 }
             } else if (isLongitude(col)) {
                 lonSpelling = col;
-                if (CSVStrategySupport.isNumeric(type)) {
+                if (CSVStrategy.isNumeric(type)) {
                     validLon = true;
                 }
             }
         }
         if (validLat && validLon) {
-            builder.add(GEOMETRY_COLUMN, Point.class);
+        	List<String> csvHeaders = Arrays.asList(headers);
+        	int index = csvHeaders.indexOf(latSpelling);
+        	AttributeTypeBuilder builder2 = new AttributeTypeBuilder();
+        	builder2.setCRS(DefaultGeographicCRS.WGS84);
+        	builder2.binding(Point.class);       	
+        	AttributeDescriptor descriptor = builder2.buildDescriptor(GEOMETRY_COLUMN);
+        	builder.add(index, descriptor);
+			
```

Comment:
Yeah sorry for the awkward workaround required here, I think we can improve the builder to be more helpful here.

How about this for a proposed API.
```
AttributeTypeBuilder builder2 = builder.get(GEOMETRY_COLUMN);
builder2.setCRS(DefaultGeographicCRS.WGS84);
builder2.binding(Point.class);
builder.set(index, builder2.buildDescriptor(GEOMETRY_COLUMN));
```